src/Ninject.Web.AspNetCore/AspNetCoreKernel.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Ninject.Activation;
+using Ninject.Modules;
+using Ninject.Parameters;
+using Ninject.Planning.Bindings;
+using System;
+using System.Collections.Generic;
+
+namespace Ninject.Web.AspNetCore
+{
+	public class AspNetCoreKernel : StandardKernel
+	{
+		public AspNetCoreKernel(params INinjectModule[] modules)
+			: base(modules) { }
+
+		public AspNetCoreKernel(INinjectSettings settings, params INinjectModule[] modules)
+			: base(settings, modules) { }
+
+		protected override Func<IBinding, bool> SatifiesRequest(IRequest request)
+		{
+			return binding => {
+				var latest = true;
+				if (request.IsUnique && request.Constraint == null)
+				{
+					latest = binding.Metadata.Get<BindingIndex.Item>(nameof(BindingIndex))?.IsLatest ?? true;
+				}
+				return binding.Matches(request) && request.Matches(binding) && latest;
```

Comment:
why not calling here the base.SatifiesRequest? I.e. base.SatifiesRequest && latest.

Reply:
The method returns a `Func<IBinding, bool>`, so I would have to return a function that calls the base function and adds the condition. Turns out, recursive `Func` calls are not particularly efficient:

```
using System.Diagnostics;

void Measure(Action action)
{
    var watch = new Stopwatch();
    watch.Start();
    action();
    watch.Stop();
    watch.Elapsed.Dump();
}

var numbers = Enumerable.Range(0, 100000000);

Func<int, bool> condition1 = x => x % 17 == 0;
Func<int, bool> condition2 = x => x % 73 == 0;
Func<int, bool> condition3 = x => x % 59 == 0;
Func<int, bool> condition4 = x => x % 31 == 0;

Measure(() => numbers.Where(x => condition1(x) && condition2(x) && condition3(x) && condition4(x)).Count().Dump());

condition1 = x => x % 17 == 0;
condition2 = x => condition1(x) && x % 73 == 0;
condition3 = x => condition2(x) && x % 59 == 0;
condition4 = x => condition3(x) && x % 31 == 0;

Measure(() => numbers.Where(x => condition4(x)).Count().Dump());

condition1 = x => x % 17 == 0 && x % 73 == 0 && x % 59 == 0 && x % 31 == 0;

Measure(() => numbers.Where(x => condition1(x)).Count().Dump());
```

Reply:
Thanks for the explanation 👍